src/ripple/net/impl/RPCCall.cpp

```diff
@@ -151,6 +151,36 @@ class RPCParser
         return v;
     }
 
+    Json::Value parseDownloadShard(Json::Value const& jvParams)
+    {
+        Json::Value v (Json::objectValue);
+
+        if (!jvParams.isMember(jss::shards) ||
+            !jvParams[0u].isArray() ||
+            jvParams[0u].size() == 0)
+        {
+            return rpcError(rpcINVALID_PARAMS);
+        }
```

Comment:
I tried invoking `download_shard` on the command line, and I think this check is wrong.

First of all, the `commands[]` array on 1099 is already checking that there is at least one argument, so if I provide 0, it doesn't even get this far. Second, if there are arguments, I don't think `jvParams[jss::shards]` will ever be set.

Examples:
```
eah@badger:~/dev/rippled-merge (miguelportilla--shard_transfer=)$ rippled download_shard
Loading: "/home/eah/.config/ripple/rippled.cfg"
2018-Jul-09 23:14:28.490922982 HTTPClient:NFO Connecting to 127.0.0.1:5015

{
   "result" : {
      "error" : "invalidParams",
      "error_code" : 31,
      "error_message" : "Missing field 'shards'.",
      "request" : {
         "command" : "download_shard"
      },
      "status" : "error"
   }
}
eah@badger:~/dev/rippled-merge (miguelportilla--shard_transfer=)$ rippled download_shard 1
Loading: "/home/eah/.config/ripple/rippled.cfg"
2018-Jul-09 23:14:37.181363866 HTTPClient:NFO Connecting to 127.0.0.1:5015

{
   "result" : {
      "error" : "invalidParams",
      "error_code" : 31,
      "error_message" : "Missing field 'shards'.",
      "request" : {
         "command" : "download_shard",
         "params" : [ "1" ]
      },
      "status" : "error"
   }
}
eah@badger:~/dev/rippled-merge (miguelportilla--shard_transfer=)$ rippled download_shard 1 https://dl.dropboxusercontent.com/s/bwfu5q3l1301hmk/1.tar.lz4
Loading: "/home/eah/.config/ripple/rippled.cfg"
2018-Jul-09 23:15:19.958912395 HTTPClient:NFO Connecting to 127.0.0.1:5015

{
   "result" : {
      "error" : "invalidParams",
      "error_code" : 31,
      "error_message" : "Missing field 'shards'.",
      "request" : {
         "command" : "download_shard",
         "params" : [ "1", "https://dl.dropboxusercontent.com/s/bwfu5q3l1301hmk/1.tar.lz4" ]
      },
      "status" : "error"
   }
}
eah@badger:~/dev/rippled-merge (miguelportilla--shard_transfer=)$ rippled download_shard 1 https://dl.dropboxusercontent.com/s/bwfu5q3l1301hmk/1.tar.lz4 validate
Loading: "/home/eah/.config/ripple/rippled.cfg"
2018-Jul-09 23:15:24.651213680 HTTPClient:NFO Connecting to 127.0.0.1:5015

{
   "result" : {
      "error" : "invalidParams",
      "error_code" : 31,
      "error_message" : "Missing field 'shards'.",
      "request" : {
         "command" : "download_shard",
         "params" : [
            "1",
            "https://dl.dropboxusercontent.com/s/bwfu5q3l1301hmk/1.tar.lz4",
            "validate"
         ]
      },
      "status" : "error"
   }
}
```

Reply:
@ximinez thanks for testing and finding that, fixed in the latest commit.